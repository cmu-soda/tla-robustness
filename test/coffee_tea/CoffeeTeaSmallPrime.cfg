SPECIFICATION Spec
INVARIANT TypeOK
INVARIANT WaterSafeTemp
