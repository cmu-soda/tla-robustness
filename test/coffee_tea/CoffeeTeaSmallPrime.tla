--------------------------- MODULE CoffeeTeaSmallPrime ---------------------------

EXTENDS Naturals

VARIABLES brewed, temp

vars == <<brewed, temp>>

Init == brewed = {} /\ temp = 70

BrewCoffee ==
    /\ brewed = {}
    /\ brewed' = {"coffee"}
    /\ temp' = 212

BrewTea ==
    /\ brewed = {}
    /\ brewed' = {"tea"}
    /\ temp' \in {212, 225}

Next ==
    \/ BrewCoffee
    \/ BrewTea
    
Spec == Init /\ [][Next]_vars

WaterSafeTemp == temp <= 220

TypeOK ==
    /\ brewed \in SUBSET {"coffee", "tea"}
    /\ temp \in Nat

BrewOneItem == \A i1,i2 \in brewed : i1 = i2

=============================================================================
\* Modification History
\* Last modified Thu Feb 09 16:05:27 EST 2023 by idardik
\* Created Thu Feb 09 15:03:34 EST 2023 by idardik

