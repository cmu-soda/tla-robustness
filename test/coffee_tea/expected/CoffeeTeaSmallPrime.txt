TLA+ Module: CoffeeTeaSmallPrime
Diff rep grouped by action:

BrewTea:
Safety boundary representation: out/CoffeeTeaSmallPrime_BrewTea_diff_rep.txt
Safety boundary rep formula (with sort definitions):
/\ brewed = {}
/\ temp = 70
